src/protocol/payload/mod.rs

```diff
@@ -74,33 +80,29 @@ impl std::ops::Deref for VarInt {
     }
 }
 
-impl VarInt {
-    fn encode(&self, buffer: &mut Vec<u8>) -> io::Result<usize> {
```

Comment:
Note the removal of `<usize>` -- required to meet the Trait definitation, and I couldn't find a use case for this (?).

Reply:
Likely left over from calculating the message length for the header, it's not needed anymore 👌